underscore.js

```diff
@@ -122,6 +122,13 @@
     return result;
   };
 
+	// An internal function to resolve object from string of properties.
+	var resolveObject = function(obj, path) {
+		return [obj || self].concat(path.split('.')).reduce(function(prev, curr) {
+			return prev[curr];
```

Comment:
This would throw on `resolveObject({ }, 'foo.bar')`. Perhaps `return prev && prev[curr]`?